CHANGELOG.md

```diff
@@ -14,6 +14,7 @@
 * [ENHANCEMENT] Alertmanager: Add receiver validations for msteamsv2 and rocketchat. #6606
 * [ENHANCEMENT] Query Frontend: Add a `-frontend.enabled-ruler-query-stats` flag to configure whether to report the query stats log for queries coming from the Ruler. #6504
 * [ENHANCEMENT] OTLP: Support otlp metadata ingestion. #6617
+* [ENHANCEMENT] Querier: Add a fallback logic when use a thanos promql engine. #6630
```

Comment:
I wouldn't call it an enhancement. It is the behavior change from thanos promql engine itself so we handle it in Cortex. There is no user facing changes.